tracing/src/level_filters.rs

```diff
@@ -95,6 +95,14 @@ impl PartialOrd<LevelFilter> for Level {
     }
 }
 
+impl LevelFilter {
+    /// Attempts to convert `self` into a [`Level`](../struct.Level.html) unless
+    /// `self` is [`OFF`](#associatedconstant.OFF).
+    pub const fn to_level(self) -> Option<Level> {
```

Comment:
Per my understanding of the Rust API guidelines [suggestions for naming](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv), I _think_ this ought to be
```suggestion
    pub const fn into_level(self) -> Option<Level> {
```

Reply:
So from what I've seen in the ecosystem, `to` either implies that 1) work might be done or 2) an operation might fail (`None` here).

That said, after reading the suggestions, I think `into` makes more sense here.

Reply:
>  2) an operation might fail (`None` here).

In this case, I wouldn't consider returning `None` a "failure" exactly? But, I'm willing to be convinced otherwise here.

Reply:
`LevelFilter::OFF` is not itself a `Level` and so it can't be converted to one. Semantically, the operation converting it into a `Level` would fail and so it returns `None`. That the inner value gets moved out here is an implementation detail. Might help to think about doing this with `match`ing.

Reply:
I still don't think about that as modeling something that "fails". To me, the option is modeling "a value that may or may not be present", and the absence of such a value is not a "failure" IMO. If the level filter is OFF, there is no corresponding `Level`; you could argue that returning `None` is successfully indicating that no corresponding `Level` exists. As an illustration, consider a language like Java or Python, where failures are indicated by exceptions rather than return values. In  those languages, I would expect a similar function to return `null` (or `nil`, however the locals pronounce it), rather than throwing an exception. 

This is probably an excessive amount of hair splitting, though.  :)